### ISSUE TITLE
v3 gives `Module not found: Error: Package path [somepath] is not exported from package`

### ISSUE THREAD

**Issue by simPod** [date-fns/date-fns#3612]
```07:47:45       ERROR in ../.yarn/__virtual__/date-fns-tz-virtual-ab571865a2/2/yarn/global/cache/date-fns-tz-npm-2.0.0-9b7996f292-10c0.zip/node_modules/date-fns-tz/esm/format/index.js 1:0-52
07:47:45       Module not found: Error: Package path ./format/index.js is not exported from package /yarn/global/cache/date-fns-npm-3.0.0-6af85aded7-10c0.zip/node_modules/date-fns/ (see exports field in /yarn/global/cache/date-fns-npm-3.0.0-6af85aded7-10c0.zip/node_modules/date-fns/package.json)
07:47:45        @ ../.yarn/__virtual__/date-fns-tz-virtual-ab571865a2/2/yarn/global/cache/date-fns-tz-npm-2.0.0-9b7996f292-10c0.zip/node_modules/date-fns-tz/esm/index.js 3:0-53 3:0-53
07:47:45       
07:47:45       ERROR in ../.yarn/__virtual__/date-fns-tz-virtual-ab571865a2/2/yarn/global/cache/date-fns-tz-npm-2.0.0-9b7996f292-10c0.zip/node_modules/date-fns-tz/esm/formatInTimeZone/index.js 1:0-60
07:47:45       Module not found: Error: Package path ./_lib/cloneObject/index.js is not exported from package /yarn/global/cache/date-fns-npm-3.0.0-6af85aded7-10c0.zip/node_modules/date-fns/ (see exports field in /yarn/global/cache/date-fns-npm-3.0.0-6af85aded7-10c0.zip/node_modules/date-fns/package.json)
07:47:45        @ ../.yarn/__virtual__/date-fns-tz-virtual-ab571865a2/2/yarn/global/cache/date-fns-tz-npm-2.0.0-9b7996f292-10c0.zip/node_modules/date-fns-tz/esm/index.js 4:0-73 4:0-73
07:47:45       
07:47:45       ERROR in ../.yarn/__virtual__/date-fns-tz-virtual-ab571865a2/2/yarn/global/cache/date-fns-tz-npm-2.0.0-9b7996f292-10c0.zip/node_modules/date-fns-tz/esm/toDate/index.js 1:0-56
07:47:45       Module not found: Error: Package path ./_lib/toInteger/index.js is not exported from package /yarn/global/cache/date-fns-npm-3.0.0-6af85aded7-10c0.zip/node_modules/date-fns/ (see exports field in /yarn/global/cache/date-fns-npm-3.0.0-6af85aded7-10c0.zip/node_modules/date-fns/package.json)
07:47:45        @ ../.yarn/__virtual__/date-fns-tz-virtual-ab571865a2/2/yarn/global/cache/date-fns-tz-npm-2.0.0-9b7996f292-10c0.zip/node_modules/date-fns-tz/esm/index.js 6:0-53 6:0-53
07:47:45       
07:47:45       ERROR in ../.yarn/__virtual__/date-fns-tz-virtual-ab571865a2/2/yarn/global/cache/date-fns-tz-npm-2.0.0-9b7996f292-10c0.zip/node_modules/date-fns-tz/esm/toDate/index.js 2:0-100
07:47:45       Module not found: Error: Package path ./_lib/getTimezoneOffsetInMilliseconds/index.js is not exported from package /yarn/global/cache/date-fns-npm-3.0.0-6af85aded7-10c0.zip/node_modules/date-fns/ (see exports field in /yarn/global/cache/date-fns-npm-3.0.0-6af85aded7-10c0.zip/node_modules/date-fns/package.json)
07:47:45        @ ../.yarn/__virtual__/date-fns-tz-virtual-ab571865a2/2/yarn/global/cache/date-fns-tz-npm-2.0.0-9b7996f292-10c0.zip/node_modules/date-fns-tz/esm/index.js 6:0-53 6:0-53
07:47:45       
07:47:45       ERROR in ../.yarn/__virtual__/date-fns-tz-virtual-ab571865a2/2/yarn/global/cache/date-fns-tz-npm-2.0.0-9b7996f292-10c0.zip/node_modules/date-fns-tz/esm/zonedTimeToUtc/index.js 1:0-60
07:47:45       Module not found: Error: Package path ./_lib/cloneObject/index.js is not exported from package /yarn/global/cache/date-fns-npm-3.0.0-6af85aded7-10c0.zip/node_modules/date-fns/ (see exports field in /yarn/global/cache/date-fns-npm-3.0.0-6af85aded7-10c0.zip/node_modules/date-fns/package.json)
07:47:45        @ ../.yarn/__virtual__/date-fns-tz-virtual-ab571865a2/2/yarn/global/cache/date-fns-tz-npm-2.0.0-9b7996f292-10c0.zip/node_modules/date-fns-tz/esm/index.js```

**Comment by alihuber** [comment]
I'm getting this error too, I'm also using `"type: "module"` in my package.json.  
File 1 is importing `import { subDays } from 'date-fns/subDays.js';` , another file has the exact same imports in it (and a few others from date-fns), but there is only one error, coming from one of the files:
`Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './subDays.js' is not defined by "exports" in /my/project/node_modules/date-fns/package.json imported from /my/own/file.js`

**Comment by RUSHt** [comment]
Node v18.18.0. "type": "module" in package.json

```
import { format } from 'date-fns';
import { utcToZonedTime } from 'date-fns-tz'
```

v3.0.0 fails with:-
```
Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './format/index.js' is not defined by "exports" in /<project>/node_modules/date-fns/package.json imported from /<project>/node_modules/date-fns-tz/esm/format/index.js
```
v2.30.0 works fine.

**Comment by ctrlaltdylan** [comment]
Also seeing this error on the latest package version `v3.0.1`.

Reverting to `v2.30.0` seems to do the trick. Thanks @RUSHt .

**Comment by kossnocorp** [comment]
`date-fns-tz` is currently incompatible with date-fns v3, you can follow the progress here: https://github.com/marnusw/date-fns-tz/issues/260

**Comment by marsilinou97** [comment]
Had the same issue, fixed it by downgrading `"date-fns": "^3.0.6" to  `"date-fns": "2.30.0"`
